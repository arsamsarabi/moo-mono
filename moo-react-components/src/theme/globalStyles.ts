import { createGlobalStyle, css } from 'styled-components';
import { hsl } from 'polished';

import type { WithTheme } from '../types';
import { resetCss } from './resetCss';
import { colors } from './palette';

const { black, white } = colors;

export const GlobalStyles = createGlobalStyle(
  ({ theme: { typography } }: WithTheme) => {
    return css`
      ${resetCss}

      body {
        margin: 0;
        padding: 0;
        font-family: ${typography.fontFamily};
        -webkit-font-smoothing: antialiased;
        -moz-osx-font-smoothing: grayscale;
        color: ${hsl(black.h, black.s, black.l)};
        background-color: ${hsl(white.h, white.s, white.l)};
        font-size: ${typography.fontSize};
      }
    `;
  },
);
