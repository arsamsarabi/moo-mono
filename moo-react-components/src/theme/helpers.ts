import type { HSL } from '../types';

export const getColourStringFromHSL = (hsl: HSL, alpha = 1): string => {
  const a = alpha < 0 ? 0 : alpha > 1 ? 1 : alpha;
  return `hsla(${hsl.h} ${hsl.s}% ${hsl.l}% ${a})`;
};
