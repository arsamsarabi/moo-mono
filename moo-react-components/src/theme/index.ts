export { theme as defaultTheme } from './theme';
export { ThemeProvider } from './ThemeProvider';
export { colors } from './palette';
