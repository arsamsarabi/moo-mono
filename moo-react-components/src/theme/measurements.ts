import type { MeasurementsType } from '../types';

export const measurements: MeasurementsType = {
  xsmall: '4px',
  small: '8px',
  medium: '16px',
  large: '32px',
};
