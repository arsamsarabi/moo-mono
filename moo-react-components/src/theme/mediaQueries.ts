const size = {
  largeMobile: '414px',
  tablet: '834px',
  laptop: '1024px',
  desktop: '2560px',
};

export const devices = {
  largeMobile: `(min-width: ${size.largeMobile})`,
  tablet: `(min-width: ${size.tablet})`,
  laptop: `(min-width: ${size.laptop})`,
  desktop: `(min-width: ${size.desktop})`,
};
