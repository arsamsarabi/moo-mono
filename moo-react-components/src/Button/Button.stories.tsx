import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';

import Button, { ButtonProps } from './Button';
import { ThemeProvider } from '../theme';

export default {
  title: 'Button',
  component: Button,
  decorators: [
    (Story) => (
      <ThemeProvider>
        <Story />
      </ThemeProvider>
    ),
  ],
} as Meta;

const Template: Story<ButtonProps> = ({ color, inverse, disabled }) => {
  return (
    <Button color={color} inverse={inverse} disabled={disabled}>
      Click me!
    </Button>
  );
};

export const Example = Template.bind({});
Example.args = {
  color: 'hotpink',
  inverse: false,
  disabled: false,
};
