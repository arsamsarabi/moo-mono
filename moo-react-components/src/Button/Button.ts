import styled, { css } from 'styled-components';
import { rgba } from 'polished';

import type { WithTheme } from '../types';
import { defaultTheme } from '../theme';

const reset = css`
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  outline: none;
  background: transparent;
  text-align: inherit;
  border-radius: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
`;

const disabledStyles = css`
  pointer-events: none;
  background-color: ${rgba(defaultTheme.palette.common.black, 0.5)};
  color: ${rgba(defaultTheme.palette.common.white, 0.5)};
  border-color: ${rgba(defaultTheme.palette.common.white, 0.5)};
`;
export interface ButtonProps extends WithTheme {
  color?: string;
  inverse?: boolean;
  disabled?: boolean;
}

const Button = styled.button<ButtonProps>(
  ({
    theme: { palette, measurements },
    color = palette.brand,
    inverse = false,
    disabled = false,
  }) => {
    return css`
      ${reset};
      border: 1px solid ${color};
      padding: ${measurements.xsmall};
      color: ${color};
      border-radius: 4px;
      transition: 0.2s all linear;
      background-color: transparent;
      text-align: center;

      &:hover,
      &:focus {
        cursor: pointer;
        background-color: ${inverse ? rgba(color, 0.25) : rgba(color, 0.85)};
        color: ${inverse ? color : palette.common.white};
      }

      ${disabled ? disabledStyles : ''}
    `;
  },
);

export default Button;
