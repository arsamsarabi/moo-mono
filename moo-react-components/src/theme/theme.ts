import { colors, palette } from './palette';
import { typography } from './typography';
import { measurements } from './measurements';
import type { ThemeType } from '../types';

export const theme: ThemeType = {
  colors,
  palette,
  typography,
  measurements,
};
