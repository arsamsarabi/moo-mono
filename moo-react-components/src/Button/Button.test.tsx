import 'jest-styled-components';
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { rgba } from 'polished';

import Button, { ButtonProps } from './Button';
import { defaultTheme } from '../theme';

describe('<Button />', () => {
  let props: ButtonProps;

  beforeEach(() => {
    props = {
      color: 'hotpink',
      inverse: false,
      disabled: false,
      theme: defaultTheme,
    };
  });

  afterEach(cleanup);

  const renderComponent = () =>
    render(<Button {...props} data-testid="button-component" />);

  it('Should match snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  it('Should have disabled styles', () => {
    props.disabled = true;
    const { container, getByTestId } = renderComponent();
    const button = getByTestId('button-component');

    expect(container).toMatchSnapshot();
    expect(button).toHaveStyleRule(
      'background-color',
      rgba(defaultTheme.palette.common.black, 0.5),
    );
    expect(button).toHaveStyleRule(
      'color',
      rgba(defaultTheme.palette.common.white, 0.5),
    );
    expect(button).toHaveStyleRule(
      'border-color',
      rgba(defaultTheme.palette.common.white, 0.5),
    );
  });
});
