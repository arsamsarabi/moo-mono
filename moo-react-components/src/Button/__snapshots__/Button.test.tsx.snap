// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Should have disabled styles 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  outline: none;
  background: transparent;
  text-align: inherit;
  border-radius: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  border: 1px solid hotpink;
  padding: 4px;
  color: hotpink;
  border-radius: 4px;
  -webkit-transition: 0.2s all linear;
  transition: 0.2s all linear;
  background-color: transparent;
  text-align: center;
  pointer-events: none;
  background-color: rgba(5,73,92,0.5);
  color: rgba(245,246,250,0.5);
  border-color: rgba(245,246,250,0.5);
}

.c0:hover,
.c0:focus {
  cursor: pointer;
  background-color: rgba(255,105,180,0.85);
  color: #f5f6fa;
}

<div>
  <button
    class="c0"
    color="hotpink"
    data-testid="button-component"
    disabled=""
  />
</div>
`;

exports[`<Button /> Should match snapshot 1`] = `
.c0 {
  border: none;
  margin: 0;
  padding: 0;
  width: auto;
  overflow: visible;
  outline: none;
  background: transparent;
  text-align: inherit;
  border-radius: 0;
  color: inherit;
  font: inherit;
  line-height: normal;
  border: 1px solid hotpink;
  padding: 4px;
  color: hotpink;
  border-radius: 4px;
  -webkit-transition: 0.2s all linear;
  transition: 0.2s all linear;
  background-color: transparent;
  text-align: center;
}

.c0:hover,
.c0:focus {
  cursor: pointer;
  background-color: rgba(255,105,180,0.85);
  color: #f5f6fa;
}

<div>
  <button
    class="c0"
    color="hotpink"
    data-testid="button-component"
  />
</div>
`;
